File: backend2/backend/ml-pipeline/pipelines/rewrite_resume.js
import { streamLLM } from '../integrations/llm.js';
import { loadPrompt } from '../utils/load_prompt.js';

export async function rewriteResume(resume, targetSkills) {
  const prompt = loadPrompt('rewrite_resume.md', {
    resume,
    target_skills: JSON.stringify(targetSkills, null, 2),
  });
  return streamLLM(prompt, { maxTokens: 8096 });
}

File: services/tailored_resume_generation/ml_pipeline/pipelines/rewrite_resume.py
import json

from ..integrations.llm import stream_llm
from ..utils.load_prompt import load_prompt


def rewrite_resume(resume: str, target_skills: dict) -> str:
    prompt = load_prompt("rewrite_resume.md", {
        "resume": resume,
        "target_skills": json.dumps(target_skills, indent=2),
    })
    return stream_llm(prompt, max_tokens=8096)
